Add guide for new editor (#259)
obsidian-community/obsidian-hub: Editing notes using the github.dev editor.md

---
aliases: 
- 
tags:
- seedling
publish: true
---

# Editing notes using the github.dev editor

%% Add a description below this line. It doesn't need to be long: one or two sentences should be a good start. %%

This guide will cover the basics of using the github.dev editor for the [[Obsidian Hub]]. This editor is great when you are making quick edits that don't need any Obsidian-specific functionality.

## Opening the github.dev editor

> To open the editor you'll usually have two options: 

Reviewer: I'm guessing people need to have a GitHub account, and be logged in to GitHub in their browser, for this to work???

Reviewer: I've added a note about that.

> 1. On the Hub's publish site, each note has an "Edit in GitHub" button in it's footer

Reviewer: Maybe worth linking to the Hub publish site and the Hub repo from these two bullet points... Many will not know what they mean, I feel.

Reviewer: I've added those links.

4. To submit your files, you have to press the <kbd>✓</kbd> button at the top. 
	The first time you make an edit to the hub, you'll get a message explaining that GitHub will create a copy of the repository on your account. This is called a **Fork**.
	![[github.dev editor commit changes.png]]
5. Next, it will ask you to give a name to the *branch* where you want to save your changes. You can choose a new name for it, or just press enter. ^8cdfd9
	![[github.dev editor create branch.png]] 
6. GitHub will let you know it will switch now to the copy of the repository in your account. And once it reloads you'll see that your changes have been saved.
	![[github.dev switch to fork.png|300]]
	![[github.dev fork view.png]]

## Submitting changes to the Hub

Using git terminology, this is called creating a **Pull Request**.

1. Clicking on [this](https://github.com/obsidian-community/obsidian-hub/compare) link is equivalent to going to the Hub repository, clicking on the Pull Requests tab and choosing New. Click on compare across forks, and choose yours on head repository. For the branch, switch to the one you chose on [[#^8cdfd9|step 6]] on the previous section:

> 	![[github compare forks.png]]

Reviewer: The info in the pictures is great. I think we should review them after the page is published, for readability and accessibility.

And if we do re-do the images with smaller windows and large font sizes, maybe we could use files from the Hub, for context?

Here's an example of an existing page, on the Publish site, so show the type of problem:

![image](https://user-images.githubusercontent.com/4840096/150657133-138f1c07-75f5-4711-a073-8c806e0cab64.png)

Reviewer 2: If need be, these images can also be opened at full size with right click, open in new window. As you said below, it is something we can improve later.

---
aliases: 
- 
tags:
- seedling
publish: true
---

# Editing notes using the github.dev editor

%% Add a description below this line. It doesn't need to be long: one or two sentences should be a good start. %%

This guide will cover the basics of using the github.dev editor for the [[Obsidian Hub]]. This editor is great when you are making quick edits that don't need any Obsidian-specific functionality.

> In particular, if you are browsing the Hub and you see a note that you would like to improve, this guide talks you through how to make the edit, using the "Edit in GitHub" button at the bottom of the note.

Reviewer 3: the guide "talks you through" sounds somewhat weird in my ears, but I am not sure how I'd rephrase it. Maybe sth like "will show you"?

> In particular, if you are browsing the Hub and you see a note that you would like to improve, this guide talks you through how to make the edit, using the "Edit in GitHub" button at the bottom of the note.

Reviewer 2: ```suggestion
In particular, if you are browsing the Hub and you see a note that you would like to improve, this guide shows you how to make the edit, using the "Edit in GitHub" button at the bottom of the note.
```

## Opening the github.dev editor

To open the editor you'll usually have two options: 
1. On the Hub's [publish site](https://publish.obsidian.md/hub/00+-+Start+here), each note has an "Edit in GitHub" button in its footer.
2. When you are in the Hub's [GitHub repository](https://github.com/obsidian-community/obsidian-hub), pressing <kbd>.</kbd> will take you to the editor. 

Note that you will need a GitHub account to make edits, and if you aren't logged in to it, launching the github.dev editor will prompt you for your GitHub login details.

## Making changes to the notes

1. To make changes, you can select files from the file explorer (1), and make edits to the note (2) as you would on any text editor or paste any changes made in Obsidian to existing notes. To add new files, you can also drag from your system's file explorer to the file explorer in the editor (1).
	![[github.dev editor.png]]
2. Once you are ready to submit your changes, switch to the version control tab. You'll see the names of the files you've modified under the "Changes" section:
    ![[github.dev editor version control.png|300]]

> 3. Add the files you want to submit by pressing the <kbd>+</kbd> sign (2), and write a message describing the changes you've made(1): 

Reviewer 3: Like, I know that there is sometimes a reason for not adding everything to a commit when working with git in general. But for the occasional contributor not familiar with git, would there ever be a scenario where you do make changes that you do *not* end up adding to the commit?

I think it would be easier to simply phrase instruct people to simply add everything (or use such an add everything button, if there is one).

Reviewer 2: In VSCode, there is a `+` button to add them all, I wouldn't go into too much detail which could be confusing.

	![[github.dev editor stage changes.png|300]]
4. To submit your files, you have to press the <kbd>✓</kbd> button at the top. 
	The first time you make an edit to the hub, you'll get a message explaining that GitHub will create a copy of the repository on your account. This is called a **Fork**.
	![[github.dev editor commit changes.png]]

> 5. Next, it will ask you to give a name to the *branch* where you want to save your changes. You can choose a new name for it, or just press enter. ^8cdfd9

Reviewer 3: for the less git experienced, I'd probably add/simplify this. Particularly for small contributions, the name the new branch gets is pretty much irrelevant, and the instruction "you can choose a new name" adds unnecessary complexity.

Reviewer 2: It says that also enter can be pressed, so that should be alright.

	![[github.dev editor create branch.png]] 
6. GitHub will let you know it will switch now to the copy of the repository in your account. And once it reloads you'll see that your changes have been saved.
	![[github.dev switch to fork.png|300]]
	![[github.dev fork view.png]]

At this stage, your edits are in your fork of the Hub.

## Submitting changes to the Hub

Now you need to submit the changes in Fork to the original Hub repository.

> Using git terminology, this is called creating a **Pull Request**.

Reviewer 3: aren't Pull requests strictly speaking git*hub* terminology? :P

Reviewer 2: ```suggestion
This is called creating a **Pull Request**.
```

1. Clicking on [this](https://github.com/obsidian-community/obsidian-hub/compare) link is equivalent to going to the Hub repository, clicking on the Pull Requests tab and choosing New. Click on compare across forks, and choose yours on head repository. For the branch, switch to the one you chose on [[#^8cdfd9|step 6]] on the previous section:
	![[github compare forks.png]]
2. Once you have selected your fork and branch, you can choose <kbd>Create pull request</kbd>
	![[github create pull request.png]] 

> 3. Fill out the template and when you are done, press on <kbd>Create pull request</kbd>.  

Reviewer 3: for the less git experienced, it will be confusing why you have to describe your changes again, after you already described your changes above in the commit message 🤔

Reviewer 2: It already adds the title from the commit message. If a second or third line was used (I think not possible with this interface), then that would show up there.
And the contributor didn't describe what changes they made before, did they?

	![[github fill out PR template.png]]

## What happens next?

Your Pull Request will be added to the [list of Pull Requests](https://github.com/obsidian-community/obsidian-hub/pulls) in the original Hub repository.

> Your edits will later be reviewed by a volunteer of the Hub project, and then merged in to the latest version of the Hub.

Reviewer 3: you might also add that the reviewer might request changes before approval

> Your edits will later be reviewed by a volunteer of the Hub project, and then merged in to the latest version of the Hub.

Reviewer 2: ```suggestion
Your edits will later be reviewed by a volunteer of the Hub project, and then merged in to the latest version of the Hub.
Reviewers may request changes before merging. 
```
